Scope config option to isolate translations

vipera opened the pull request:
In my usage of this gem, I ran into a situation where I'd like two very related applications that are organizationally distinct but part of the same domain (e.g. public web application and its admin/management application) to connect to the same database.

However, I'd like them *not* to share translation data, and found that difficult to do without overriding methods in `I18n::Backend::ActiveRecord`. So instead, I added this feature to my fork, and wanted to see what you think about it and whether you'd be interested in having it in the main codebase.

Essentially I added a `scope` field to the migration and if the configuration option is set, then all reads/writes will be limited to only those records with that particular scope identifier. I wanted to maintain compatibility with existing tables that don't have a scope, or even newly created tables where someone wants to remove the scope field from the migration if they won't be using it.

I've added two tests showing how I think the behaviour should work and a readme paragraph. Thanks taking the time to look this over :relaxed:

vipera committed:
Add scope config option

Allow the entire backend to use translations only from a subset of
the translation records.

timfjord commented:
It is looking good 👍🏻, there are some failed tests though

vipera commented:
Thanks for your input, Tim!

I glanced over the pipeline logs and it looks like tests are failing on the master branch too for the same reasons. As usual, test pipelines stop being reliable after a while...

I can look into the details to try and fix it, but I don't think it's related to the change I made.

timfjord commented:
Hm, indeed, `master` is broken.
It would great to fix the tests(if you have bandwidth of course) otherwise I will try to fix `master` so we can go ahead with this is PR

vipera committed:
Improve CI

- Add Rails 8 to test matrix
- Add excludes for incompatable Ruby/Rails combinations
- Fix sqlite3 dependencies for older Rails versions

vipera commented:
@timfjord I've rebased onto the latest master, but the CI pipeline had issues again so I went ahead and made some changes. It was mostly just an issue with sqlite3, which needed to be fixed to a specific version once again.

While I was at it, I added Ruby 3.3 and Rails 8 to the matrix. For Ruby 3.3, I got an error I don't really understand for Rails-head, so I excluded it from the matrix. Here's the error from the [failing pipeline](https://github.com/vipera/i18n-active_record/actions/runs/12234882753/job/34124987494) in case you want to take a look. Since it's head, it could change with the next commit, and I thought having a working pipeline for now might be the better option.

```
/home/runner/work/i18n-active_record/i18n-active_record/test/active_record_test.rb:177:in '<class:WithCustomTranslationModel>': uninitialized constant I18n::Backend::ActiveRecord (NameError)

    class CustomTranslation < I18n::Backend::ActiveRecord::Translation
                                           ^^^^^^^^^^^^^^
	from /home/runner/work/i18n-active_record/i18n-active_record/test/active_record_test.rb:176:in '<class:I18nBackendActiveRecordTest>'
	from /home/runner/work/i18n-active_record/i18n-active_record/test/active_record_test.rb:5:in '<top (required)>'
	from /home/runner/.rubies/ruby-head/lib/ruby/3.4.0+1/bundled_gems.rb:83:in 'Kernel.require'
	from /home/runner/.rubies/ruby-head/lib/ruby/3.4.0+1/bundled_gems.rb:83:in 'block (2 levels) in Kernel#replace_require'
	from /home/runner/work/i18n-active_record/i18n-active_record/vendor/bundle/ruby/3.4.0+1/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:21:in 'block in <main>'
	from /home/runner/work/i18n-active_record/i18n-active_record/vendor/bundle/ruby/3.4.0+1/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in 'Array#select'
	from /home/runner/work/i18n-active_record/i18n-active_record/vendor/bundle/ruby/3.4.0+1/gems/rake-13.2.1/lib/rake/rake_test_loader.rb:6:in '<main>'
```

Other than that, I made almost no other changes, so I'd appreciate a merge if you're still satisfied with the implementation 😃

timfjord commented:
Released as [v1.4.0](https://github.com/svenfuchs/i18n-active_record/releases/tag/v1.4.0)
